### PR TITLE
Remove all stray prints in tests.

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: app_flutter/lib/build_dashboard_page.dart [hunk 1]
```diff
@@ -13,6 +13,17 @@ import 'status_grid.dart';
 
 /// [BuildDashboard] parent widget that manages the state of the dashboard.
 class BuildDashboardPage extends StatefulWidget {
+  // TODO(ianh): there's a number of problems with the design here
```
Comment (CaseyHillers) [inline app_flutter/lib/build_dashboard_page.dart]: Same with this one.

File: app_flutter/lib/index_page.dart [hunk 2]
```diff
@@ -12,6 +12,15 @@ import 'sign_in_button.dart';
 import 'state/index.dart';
 
 class IndexPage extends StatefulWidget {
+  // TODO(ianh): there's a number of problems with the design here
```
Comment (CaseyHillers) [inline app_flutter/lib/index_page.dart]: Same TODO comment here.

File: app_flutter/lib/state/brooks.dart [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import 'package:flutter/foundation.dart';
+
+typedef BrookCallback<T> = void Function(T event);
+
+/// A source of events of type T.
+///
+/// This is a light-weight stream, hence the name of the class.
+///
+/// Listeners cannot be registered multiple times simultaneously.
+class Brook<T> {
```
Comment (CaseyHillers) [inline app_flutter/lib/state/brooks.dart]: This is a clever name
brook: a small stream

File: app_flutter/lib/state/brooks.dart [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  Brook();
+
+  final Set<BrookCallback<T>> _listeners = <BrookCallback<T>>{};
```
Comment (CaseyHillers) [inline app_flutter/lib/state/brooks.dart]: Just a note: I don't think there's any current use cases in app_flutter where there are multiple listeners

File: app_flutter/lib/state/flutter_build.dart [hunk 5]
```diff
@@ -11,6 +11,8 @@ import 'package:cocoon_service/protos.dart' show Commit, CommitStatus, RootKey,
 import '../service/cocoon.dart';
 import '../service/google_authentication.dart';
 
+import 'brooks.dart';
```
Comment (CaseyHillers) [inline app_flutter/lib/state/flutter_build.dart]: Should this be grouped in the above import block? Or do imports in the CWD get grouped in their own import block?

File: app_flutter/test/agent_dashboard_page_test.dart [hunk 6]
```diff
@@ -77,39 +80,56 @@ void main() {
     await tester.tap(find.text('Create Agent'));
     await tester.pump();
 
-    await tester.tap(find.text('Create'));
-    await tester.pump();
+    await checkOutput(
```
Comment (CaseyHillers) [inline app_flutter/test/agent_dashboard_page_test.dart]: Thanks for adding more test coverage in the process of this!

File: app_flutter/test/state/flutter_build_test.dart [hunk 7]
```diff
@@ -114,11 +125,18 @@ void main() {
       when(mockService.fetchTreeBuildStatus(branch: _defaultBranch))
           .thenAnswer((_) => Future<CocoonResponse<bool>>.value(CocoonResponse<bool>()..error = 'error'));
 
-      await tester.pump(buildState.refreshRate * 2);
-      verify(mockService.fetchTreeBuildStatus(branch: _defaultBranch)).called(greaterThan(1));
+      await checkOutput(
+        block: () async {
+          await tester.pump(buildState.refreshRate);
+          verify(mockService.fetchTreeBuildStatus(branch: _defaultBranch)).called(greaterThan(1));
```
Comment (CaseyHillers) [inline app_flutter/test/state/flutter_build_test.dart]: Should this be outside of checkOutput?

File: app_flutter/test/utils/fake_flutter_build.dart [hunk 8]
```diff
@@ -10,6 +10,7 @@ import 'package:cocoon_service/protos.dart' show Commit, CommitStatus, Task;
 
 import 'package:app_flutter/service/google_authentication.dart';
 import 'package:app_flutter/state/flutter_build.dart';
+import 'package:app_flutter/state/brooks.dart';
```
Comment (CaseyHillers) [inline app_flutter/test/utils/fake_flutter_build.dart]: nit: This should be above the previous import

File: app_flutter/test/utils/output.dart [hunk 9]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
```
Comment (CaseyHillers) [inline app_flutter/test/utils/output.dart]: nit: 2020

File: app_flutter/test/utils/output.dart [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+typedef AsyncVoidCallback = Future<void> Function();
+
+Future<void> checkOutput({
```
Comment (CaseyHillers) [inline app_flutter/test/utils/output.dart]: nit: Add a doc comment explaining  this checks blocks use of debugPrint

File: app_flutter/lib/agent_dashboard_page.dart [hunk 11]
```diff
@@ -14,6 +14,17 @@ import 'state/agent.dart';
 
 /// [AgentDashboardPage] parent widget that manages the state of the dashboard.
 class AgentDashboardPage extends StatefulWidget {
+  // TODO(ianh): there's a number of problems with the design here
```
[the remark this answers is not in the document]
Comment (Hixie) [reply app_flutter/lib/agent_dashboard_page.dart]: Yeah, I was going to do it in this PR but it got out of hand. I have a branch where I've started on this already.

File: app_flutter/lib/state/brooks.dart (hunk 3 above)
Comment (Hixie) [reply app_flutter/lib/state/brooks.dart]: maybe too clever, but it's buried in some random tiny codebase, so... :-)

File: app_flutter/lib/state/brooks.dart (hunk 4 above)
Comment (Hixie) [reply app_flutter/lib/state/brooks.dart]: I expect we'll have a second listener if we add logging to the server.

File: app_flutter/lib/state/flutter_build.dart (hunk 5 above)
Comment (Hixie) [reply app_flutter/lib/state/flutter_build.dart]: Good point, will fix.

File: app_flutter/test/state/flutter_build_test.dart (hunk 7 above)
Comment (Hixie) [reply app_flutter/test/state/flutter_build_test.dart]: Sure, will do.

File: app_flutter/test/utils/fake_flutter_build.dart (hunk 8 above)
Comment (Hixie) [reply app_flutter/test/utils/fake_flutter_build.dart]: fixed

File: app_flutter/test/utils/output.dart (hunk 9 above)
Comment (Hixie) [reply app_flutter/test/utils/output.dart]: I started this file by copying and pasting another so it's a derivative work of another file. I copied that file's copyright date.

File: app_flutter/test/utils/output.dart (hunk 10 above)
Comment (Hixie) [reply app_flutter/test/utils/output.dart]: added docs.

**Comment by Hixie** [comment]
Thanks for the review, will land on green (probably in the morning).

File: app_flutter/lib/state/brooks.dart (hunk 3 above)
Comment (CaseyHillers) [reply app_flutter/lib/state/brooks.dart]: It's good. I got a chuckle when I saw the definition and had to share.

**Pull request merged**